pydm/tests/widgets/test_base.py

```diff
@@ -353,6 +353,32 @@ def test_channels_for_tools(qtbot):
     assert all(x == y for x, y in zip(pydm_label.channels(), pydm_label.channels_for_tools()))
 
 
+def test_pydmwidget_channel_change(qtbot):
+    """
+    Test the channel property for changes and the effect on the channels() property.
+
+    Parameters
+    ----------
+    qtbot : fixture
+        Window for widget testing
+
+    """
+    pydm_label = PyDMLabel()
+    qtbot.addWidget(pydm_label)
+    assert pydm_label._channel is None
+    assert pydm_label._channels is None
+
+    pydm_label.channel = 'foo://bar'
+    assert pydm_label._channel == 'foo://bar'
+    assert pydm_label.channels()[0].address == 'foo://bar'
+    assert pydm_label._channels is not None
```

Comment:
If the previous assertion is True, there's no need to assert pydm_label._channels is not None. Either remove this assertion, or assert for this before asserting the _channels[0].address.